### PR TITLE
Add cases to cover pure checkpoint commands for backup function

### PR DESCRIPTION
**Pull request by lento-sun**
Signed-off-by: Yi Sun <yisun@redhat.com>

### CODE REVIEW

**Comment by lento-sun** [comment]
depend on https://github.com/avocado-framework/avocado-vt/pull/2599

**Comment by lento-sun** [comment]
Test result as follow:
```
(.libvirt-ci-venv-ci-runtest-slq4qc) [root@lenovo-sr630-14 ~]# avocado run --vt-type libvirt type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd
JOB ID     : 493637351d5d6eb5d4d8e25ba9f1411f6ca3b7b6
JOB LOG    : /root/avocado/job-results/job-2020-06-23T05.56-4936373/job.log
 (01/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-create.redefine.xml_with_domain: PASS (31.17 s)
 (02/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-create.redefine.xml_without_domain: PASS (31.02 s)
 (03/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-create-as.print-xml: PASS (30.94 s)
 (04/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-info: PASS (31.36 s)
 (05/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.no-flag: PASS (31.28 s)
 (06/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.parent: PASS (31.29 s)
 (07/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.roots: PASS (31.19 s)
 (08/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.tree: PASS (31.27 s)
 (09/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.name: PASS (31.37 s)
 (10/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.topological: PASS (31.33 s)
 (11/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.from: PASS (31.31 s)
 (12/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.descendants: PASS (31.08 s)
 (13/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.no-leaves: PASS (31.42 s)
 (14/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.leaves: PASS (30.97 s)
 (15/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-dumpxml.security-info: PASS (56.67 s)
 (16/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-dumpxml.size: CANCEL: '--size' not supported yet(bz1814573) (30.95 s)
 (17/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-parent: PASS (30.72 s)
 (18/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-create.redefine: PASS (31.94 s)
 (19/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-create-as.print-xml: PASS (31.93 s)
 (20/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-create-as.no-flag: PASS (31.81 s)
 (21/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-info.no-flag: PASS (32.57 s)
 (22/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.no-flag: PASS (32.55 s)
 (23/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.parent: PASS (32.30 s)
 (24/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.roots: PASS (32.45 s)
 (25/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.tree: PASS (32.85 s)
 (26/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.name: PASS (32.87 s)
 (27/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.topological: PASS (32.31 s)
 (28/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.from: PASS (32.86 s)
 (29/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.descendants: PASS (32.39 s)
 (30/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.no-leaves: PASS (32.49 s)
 (31/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-list.leaves: PASS (32.60 s)
 (32/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-dumpxml.security-info: PASS (32.18 s)
 (33/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-dumpxml.no-domain: PASS (32.59 s)
 (34/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-dumpxml.size: PASS (32.36 s)
 (35/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-parent: PASS (32.12 s)
 (36/36) type_specific.io-github-autotest-libvirt.incremental_backup.checkpoint_cmd.checkpoint-delete.metadata: PASS (32.27 s)
RESULTS    : PASS 35 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB TIME   : 1176.26 s
```

**Comment by chunfuwen** [comment]
Is it better put those files into virsh_cmd folder since it align with all currently virsh command?

File: libvirt/tests/src/incremental_backup/incremental_backup_checkpoint_cmd.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+        vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+        logging.debug("Script insert xml elements to make sure vm can support "
+                      "incremental backup. This should be removded when "
+                      "bz 1799015 fixed.")
+        if vm.is_alive():
+            vm.destroy(gracefully=False)
+        # Prepare the disk to be used.
+        disk_params = {}
+        disk_path = ""
+        image_name = "vdb_image.qcow2"
```
Comment (Yingshun) [inline libvirt/tests/src/incremental_backup/incremental_backup_checkpoint_cmd.py:86]: Can we update the hardcoded image_name to "{}_image.qcow2".format(test_disk_target)?

File: libvirt/tests/src/incremental_backup/incremental_backup_checkpoint_cmd.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+        disk_params = {"device_type": "disk",
+                       "type_name": "file",
+                       "driver_type": "qcow2",
+                       "target_dev": test_disk_target,
+                       "source_file": disk_path}
+        disk_xml = libvirt.create_disk_xml(disk_params)
+        virsh.attach_device(vm.name, disk_xml,
+                            flagstr="--config", **virsh_dargs)
+        vm.start()
+        session = vm.wait_for_login()
```
Comment (Yingshun) [inline libvirt/tests/src/incremental_backup/incremental_backup_checkpoint_cmd.py:99]: It seems the variant 'session' is not used, can we update this line to vm.wait_for_login().close()?

File: libvirt/tests/src/incremental_backup/incremental_backup_checkpoint_cmd.py [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+        elif checkpoint_cmd == "virsh_list":
+            stdout = virsh.dom_list(cmd_flag, **virsh_dargs).stdout.strip()
+            if ((vm_name in stdout and cmd_flag == "--without-checkpoint") or
+                    (vm_name not in stdout and cmd_flag == "--with-checkpoint")):
+                test.fail("virsh list with '%s' contains wrong data" % cmd_flag)
+    finally:
+        # Remove checkpoints
+        if "current_checkpoints" in locals() and current_checkpoints:
+            for checkpoint_name in current_checkpoints:
+                virsh.checkpoint_delete(vm_name, checkpoint_name, debug=True)
```
Comment (Yingshun) [inline libvirt/tests/src/incremental_backup/incremental_backup_checkpoint_cmd.py:263]: How about adding ignore_status=True to avoid exceptions?

Comment (lento-sun) [reply libvirt/tests/src/incremental_backup/incremental_backup_checkpoint_cmd.py]: Fixed

File: libvirt/tests/src/incremental_backup/incremental_backup_checkpoint_cmd.py (hunk 2 above)
Comment (lento-sun) [reply libvirt/tests/src/incremental_backup/incremental_backup_checkpoint_cmd.py]: fixed

File: libvirt/tests/src/incremental_backup/incremental_backup_checkpoint_cmd.py (hunk 1 above)
Comment (lento-sun) [reply libvirt/tests/src/incremental_backup/incremental_backup_checkpoint_cmd.py]: updated

File: libvirt/tests/cfg/incremental_backup/incremental_backup_checkpoint_cmd.cfg [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                - size:
+                    flag = "--size"
+                    # not supported yet, bz1814573
+                # --no-domain:
+                # This is  covered by:
+                # checkpoint_cmd.checkpoint-create.redefine.xml_without_domain.
+                # We don't test it again here.
+        - checkpoint-parent:
+            required_checkpoints = 2
+        # - checkpoint-delete:
```
Comment (chunfuwen) [inline libvirt/tests/cfg/incremental_backup/incremental_backup_checkpoint_cmd.cfg:60]: remove commented out contents

**Comment by lento-sun** [comment]
> Is it better put those files into virsh_cmd folder since it align with all currently virsh command?

This cmd won't be put in our test plan "virsh cmd - confirmed", and checkpoint operation itself is not an independent function to be tested. It's only useful in backup scenarios. So I think put it within 'incremental_backup' folder is more reasonable, what do you think?

File: libvirt/tests/cfg/incremental_backup/incremental_backup_checkpoint_cmd.cfg (hunk 4 above)
Comment (lento-sun) [reply libvirt/tests/cfg/incremental_backup/incremental_backup_checkpoint_cmd.cfg]: udpated

**Review by chunfuwen — APPROVED** [review]
LGTM

**Pull request merged**